support/azure/azure-kubernetes/availability-performance/node-auto-repair-errors.md

```diff
@@ -0,0 +1,38 @@
+
+## Prerequisites
+To determine what type of node auto-repair error has occured, you will need to look for the following Kubernetes event: 
+"Node auto-repair reboot/reimage/redeploy action failed due to an operation failure: [error code]."
+
+> [!NOTE]
+> In most cases, node auto-repair errors will not cause any impact to your cluster or applications since your node was already in an unhealthy state prior to the auto-repair process. When encountering node auto-repair errors, we recommend that you first attempt to repair the node yourself by following the instructions here: [Basic troubleshooting of Node Not Ready failures](./node-not-ready-basic-troubleshooting.md). If you are unable to restore your node back to a Succeeded status and are noticing persistent errors reported by node auto-repair, you may be encountering a rare issue and should reach out to Azure support.
+
+## Common error codes
+The table below contains the most common node auto-repair errors.
+
+| Error code | Causes & Solution |
+|---|---|
+| ARM ErrorCode: VMExtensionProvisioningError | One or more VM extensions failed to be provisioned on the VM. Read more on possible error types and troubleshooting steps at [Troubleshoot the ERR_VHD_FILE_NOT_FOUND error code (124)](../create-upgrade-delete/error-code-vhdfilenotfound.md). To determine the exact VM extension provisioning error on your node, read more at [View error details in the Azure portal](../create-upgrade-delete/troubleshoot-aks-cluster-creation-issues.md#view-resources-in-the-azure-portal). |
+| ARM ErrorCode: InvalidParameter | This occurs when the node auto-repair process requests a VM that no longer exists. [more guidance] |
+| scaleSetNameAndInstanceIDFromProviderID failed | This is an internal issue where the node may not have been provisioned correctly. If you notice any new symptoms or interruptions on your node, please reach out to Azure support. |
+| ManagedIdentityCredential authentication failed | This is likely due to an internal issue where your nodes were not initialized properly. If you notice any new symptoms or interruptions on your node, please reach out to Azure support. |
+| ARM ErrorCode: VMRedeploymentFailed | There was an error with redeploying your node. This may cause your nodepool to enter a Failed state. Read more on potential causes and next steps at [Troubleshoot Azure Kubernetes Service clusters or nodes in a failed state](./cluster-node-virtual-machine-failed-state.md#scenario-3-node-pool-is-in-a-failed-state). |
```

Comment:
```suggestion
| ARM ErrorCode: VMRedeploymentFailed | There was an error with redeploying your node. This may cause your nodepool to enter a failed state. Read more on potential causes and next steps at [Troubleshoot Azure Kubernetes Service clusters or nodes in a failed state](./cluster-node-virtual-machine-failed-state.md#scenario-3-node-pool-is-in-a-failed-state). |
```